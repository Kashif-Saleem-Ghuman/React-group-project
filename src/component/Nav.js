import React from 'react';
import { NavLink } from 'react-router-dom';
// import { FaBars } from 'react-icons';
import logo from '../logo.png';
import '../index.css';

const Nav = () => (
  <>
    <nav className="navbar hidden">
      <div className="logo">
        <img src={logo} alt="logo" />
        <h1>Space Travelers Hub</h1>
      </div>
      <ul className="nav-links">
        <li><NavLink to="/" className={({ isActive }) => (isActive ? 'active' : 'inactive')}>Rockets</NavLink></li>
        <li><NavLink to="/missions" className={({ isActive }) => (isActive ? 'active' : 'inactive')}>Missions</NavLink></li>
        <li className="myprofile"><NavLink to="/myprofile" className={({ isActive }) => (isActive ? 'active' : 'inactive')}>My Profile</NavLink></li>
      </ul>
    </nav>

    {/* <div className="mobile-popup">
      <FaBars />
    </div> */}
  </>

);

export default Nav;
