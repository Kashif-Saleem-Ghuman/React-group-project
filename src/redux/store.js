import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './slice/rocketslice';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
  },
});

export default store;
