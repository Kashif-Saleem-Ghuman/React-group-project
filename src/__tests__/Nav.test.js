/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import Nav from '../component/Nav';

describe('Nav', () => {
  it('renders Nav component', () => {
    const n = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Nav />
        </BrowserRouter>
      </Provider>,
    ).toJSON();
    expect(n).toMatchSnapshot();
  });
});
