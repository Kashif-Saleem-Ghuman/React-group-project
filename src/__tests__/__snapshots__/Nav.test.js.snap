// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav renders Nav component 1`] = `
<nav
  className="navbar hidden"
>
  <div
    className="logo"
  >
    <img
      alt="logo"
      src="logo.png"
    />
    <h1>
      Space Travelers Hub
    </h1>
  </div>
  <ul
    className="nav-links"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className="inactive"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li
      className="myprofile"
    >
      <a
        className="inactive"
        href="/myprofile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;
